File: frontend/src/types/board.ts
export interface BoardSnapshot {
  elements: unknown[];
  appState: Record<string, unknown>;
  files?: Record<string, unknown>;
}

export interface Board {
  id: string;
  title: string;
  snapshot: BoardSnapshot | null;
  createdAt: Date;
  updatedAt: Date;
  userId: string;
  workspaceId?: string; // Add workspace support
}

export interface CreateBoardInput {
  title: string;
  snapshot?: BoardSnapshot | null;
  userId: string;
  workspaceId?: string; // Add workspace support
}

export interface UpdateBoardInput {
  id: string;
  title?: string;
  snapshot?: BoardSnapshot | null;
  userId?: string;
  workspaceId?: string; // Add workspace support
}

File: backend-core-service/be-core/src/main/java/com/naammm/becore/entity/Board.java
package com.naammm.becore.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Lob;
import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "boards")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Board {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    @Column(nullable = false)
    private String title;

    /**
     * JSON payload representing the Excalidraw snapshot (elements, app state, files).
     */
    @Lob
    private String snapshot;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @Column(name = "user_id", nullable = false, length = 160)
    private String userId;

    @Column(name = "workspace_id", length = 160)
    private String workspaceId;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
